### PR TITLE
HAWQ-1035. Map sub partition table name to the table list in .yml con…

### PR DESCRIPTION
**Pull request by linwen**
Please review. Thanks!

### CODE REVIEW

File: tools/bin/hawqregister [hunk 1]
```diff
@@ -585,6 +596,20 @@ class HawqRegister(object):
...
+        mappings = []
+        pos = 0
+        for constraint in self.partitions_constraint:
+            if partitions.has_key(constraint):
+                for i in range(0, len(self.partitions_filepaths[pos])):
+                    value = (partitions[constraint], (self.partitions_filepaths[pos][i], self.partitions_filesizes[pos][i]))
+                    mappings.append(value)
+                    i += 1
+            pos += 1
+
```
Comment (xunzhang) [inline tools/bin/hawqregister]: useless newline

File: tools/bin/hawqregister [hunk 2]
```diff
@@ -585,6 +596,20 @@ class HawqRegister(object):
         query += "end transaction;"
         return self.utility_accessor.update_catalog(query)
 
+    def _mapping_tablename_from_yml(self, partitions):
```
Comment (xunzhang) [inline tools/bin/hawqregister]: It is a little like c-style python, I recommend to rewrite it as below:

```
def _mapping_tablename_from_yml(self, partitions):
    ''' Mapping table name from yml file, return a list of (table_name,(file_path, file_size)) '''
    for pos, constraint in enumerate(self.partitions_constraint):
        if partitions.has_key(constraint):
            mappings = ([(partitions[constraint], (self.partitions_filepaths[pos][i], self.partitions_filesizes[pos][i]))
                        for i in xrange(len(self.partitions_filepaths[pos]))])
    return mappings
```

Comment (linwen) [inline tools/bin/hawqregister]: Thank you!

**Comment by zhangh43** [comment]
＋1

**Pull request closed without merging**